### ISSUE TITLE
missing subnet in cloud-config causes incorrect vpc-id (aws)

### ISSUE THREAD

**Issue by thz** [kubermatic/kubermatic#4137]
Upstream provider for aws has a "dummy instance"-mode of operation for the situation of operating in an account which differs from the instance the ccm runs on. In that mode only the local metadata endpoint is using the provided VPC instead of the one queried from the local metadata endpoint.

The condition for that mode requires that either a SubnetID or a RoleARN is specified. With our recent removal of the subnet from the cloud-config the condition is no longer fulfilled. For that reason an incorrect vpc-id is used during elb creation (subnet selection).

As it is semantically correct to not specify a real subnet in the cloud-config (node-deployments, elb creation all have their own means of determining subnet), it seems valid to pass a bogus subnet-id to the cloud-config just to trigger the "dummy instance"-mode.

reference: 

https://github.com/kubernetes/kubernetes/blob/v1.14.5/pkg/cloudprovider/providers/aws/aws.go#L1231

```
	if cfg.Global.VPC != "" && (cfg.Global.SubnetID != "" || cfg.Global.RoleARN != "") && tagged {
```

**Comment by mrIncompetent** [comment]
This bug only gets triggered when setting an InstanceProfile for an AWS cluster.
Also this only occurs when using Kubermatic v2.11.
The code in `master` changed the way we handle that & is not affected.

The UI issue: https://github.com/kubermatic/dashboard-v2/issues/1484

**Comment by toschneck** [comment]
Post Mortem needed - Template: https://landing.google.com/sre/sre-book/chapters/postmortem/

**Comment by alvaroaleman** [comment]
According to @thz wontfix for 2.11 and fixed in 2.12
